### PR TITLE
More meta

### PR DESCRIPTION
**Pull request by CMGS**

### CODE REVIEW

**Comment by CMGS** [comment]
[更新组件](https://trello.com/c/GKqD5mlm/78-%E6%9B%B4%E6%96%B0%E7%BB%84%E4%BB%B6)

File: cluster/calcium/control.go [hunk 1]
```diff
@@ -87,8 +74,12 @@ func (c *Calcium) ControlContainer(ctx context.Context, IDs []string, t string,
 	return ch, nil
 }
 
-func (c *Calcium) doStartContainer(ctx context.Context, container *types.Container, containerInfo *enginetypes.VirtualizationInfo, force bool) ([]*bytes.Buffer, error) {
+func (c *Calcium) doStartContainer(ctx context.Context, container *types.Container, force bool) ([]*bytes.Buffer, error) {
 	var message []*bytes.Buffer
+	if container.Status != nil && container.Status.Running {
```
Comment (CMGS) [inline cluster/calcium/control.go:79]: 可以考虑这里 inspect

File: cluster/calcium/create.go [hunk 2]
```diff
@@ -151,6 +151,8 @@ func (c *Calcium) doCreateAndStartContainer(
 		Privileged: opts.Entrypoint.Privileged,
 		Engine:     node.Engine,
 		SoftLimit:  opts.SoftLimit,
+		Image:      opts.Image,
+		Status:     &types.ContainerStatus{},
```
Comment (CMGS) [inline cluster/calcium/create.go]: 初始化 or nil 都能使得 start 成功执行，这里可能需要搞下 control 的条件判断

File: rpc/gen/core.proto [hunk 3]
```diff
@@ -157,14 +157,12 @@ message Container {
     map<string, int32> cpu = 5;
     double quota = 6;
     int64 memory = 7;
-    bool privileged = 8;
-    map<string, string> labels = 9;
-    map<string, string> publish = 10;
-    string image = 11;
-    bytes inspect = 12;
+    int64 storage = 8;
```
Comment (CMGS) [inline rpc/gen/core.proto]: 感觉没必要做不兼容改动啊

File: tools/updatev3.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	for _, c := range cs {
+		ci, err := c.Inspect(ctx)
+		if err != nil {
+			panic(err)
+		}
+		c.Labels = ci.Labels
+		c.User = ci.User
+		c.Image = ci.Image
+		c.Env = ci.Env
+		if err := store.UpdateContainer(ctx, c); err != nil {
```
Comment (anrs) [inline tools/updatev3.go:31]: It should be updated offline (or locked) as the concurrent updating from ERU may overrides them.

File: cluster/calcium/control.go (hunk 1 above)
Comment (anrs) [reply cluster/calcium/control.go:79]: my view is the operations should be unified, all of them either with inspecting or without. Indeed, it does inspect may be further better.

File: cluster/calcium/create.go (hunk 2 above)
Comment (anrs) [reply cluster/calcium/create.go]: is it more understandable if there's a PENDING/DRAFT or something like that.

**Pull request merged**